app/scripts/metamask-controller.js

```diff
@@ -734,6 +738,20 @@ export default class MetamaskController extends EventEmitter {
         this.gasFeeController,
       ),
     });
+    this.smartTransactionsController = new SmartTransactionsController({
+      onNetworkStateChange: this.networkController.store.subscribe.bind(
+        this.networkController.store,
+      ),
+      getNetwork: this.networkController.getNetworkState.bind(
+        this.networkController,
+      ),
+      nonceTracker: this.txController.nonceTracker,
```

Comment:
I see the change made for the nonce tracker, but can you do the same for the transactions controller as well: just pass `confirmExternalTransaction` instead of the whole controller. We try not to pass around whole controllers.